service/src/PlatformTopo.cpp

```diff
@@ -423,31 +430,45 @@ namespace geopm
         PlatformTopoImp::create_cache();
     }
 
+    void PlatformTopo::create_cache_service(void)
+    {
+        PlatformTopoImp::create_cache_service();
+    }
+
     void PlatformTopoImp::create_cache(void)
     {
         PlatformTopoImp::create_cache(M_CACHE_FILE_NAME);
     }
 
+    void PlatformTopoImp::create_cache_service(void)
+    {
+        PlatformTopoImp::create_cache(M_SERVICE_CACHE_FILE_NAME);
+    }
+
     void PlatformTopoImp::create_cache(const std::string &cache_file_name)
     {
         // If cache file is not present, create it
         struct stat cache_stat;
-        if (stat(cache_file_name.c_str(), &cache_stat)) {
-            std::string cmd = "out=" + cache_file_name + ";"
-                              "lscpu -x > $out && "
-                              "chmod a+rw $out";
-            FILE *pid;
-            int err = geopm_sched_popen(cmd.c_str(), &pid);
-            if (err) {
-                unlink(cache_file_name.c_str());
-                throw Exception("PlatformTopo::create_cache(): Could not popen lscpu command: ",
-                         err, __FILE__, __LINE__);
-            }
-            if (pclose(pid)) {
-                unlink(cache_file_name.c_str());
-                throw Exception("PlatformTopo::create_cache(): Could not pclose lscpu command: ",
-                         errno ? errno : GEOPM_ERROR_RUNTIME, __FILE__, __LINE__);
-            }
+        int err = stat(cache_file_name.c_str(), &cache_stat);
+        if (err) {
+            throw Exception("PlatformTopoImp::create_cache(): stat failure:",
+                     errno ? errno : GEOPM_ERROR_RUNTIME, __FILE__, __LINE__);
+        }
+
+        std::string cmd = "out=" + cache_file_name + ";"
```

Comment:
Use move semantics: mkstemp, make a temp file, write the lscpu contents to it, then move it to cache_file_name.